algorithmic_efficiency/workloads/cifar/cifar_pytorch/workload.py

```diff
@@ -0,0 +1,248 @@
+"""CIFAR10 workload implemented in PyTorch."""
+
+import contextlib
+import math
+import os
+import random
+from typing import Dict, Tuple
+
+import torch
+from torch import nn
+import torch.distributed as dist
+import torch.nn.functional as F
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torchvision import transforms
+from torchvision.datasets import CIFAR10
+
+from algorithmic_efficiency import data_utils
+from algorithmic_efficiency import param_utils
+from algorithmic_efficiency import spec
+import algorithmic_efficiency.random_utils as prng
+from algorithmic_efficiency.workloads.cifar.workload import BaseCifarWorkload
+from algorithmic_efficiency.workloads.imagenet.imagenet_pytorch.models import \
+    resnet18
+
+PYTORCH_DDP = 'LOCAL_RANK' in os.environ
+RANK = int(os.environ['LOCAL_RANK']) if PYTORCH_DDP else 0
+DEVICE = torch.device(f'cuda:{RANK}' if torch.cuda.is_available() else 'cpu')
+N_GPUS = torch.cuda.device_count()
+
+
+class CifarWorkload(BaseCifarWorkload):
+
+  def __init__(self):
+    self._param_types = None
+    self._eval_iters = {}
+
+  @property
+  def param_shapes(self):
+    if self._param_shapes is None:
+      raise ValueError(
+          'This should not happen, workload.init_model_fn() should be called '
+          'before workload.param_shapes!')
+    return self._param_shapes
+
+  @property
+  def model_params_types(self):
+    """The shapes of the parameters in the workload model."""
+    if self._param_types is None:
+      self._param_types = param_utils.pytorch_param_types(self._param_shapes)
+    return self._param_types
+
+  def build_input_queue(self,
+                        data_rng: spec.RandomState,
+                        split: str,
+                        data_dir: str,
+                        global_batch_size: int):
+    it = self._build_dataset(data_rng, split, data_dir, global_batch_size)
+    for batch in it:
+      yield {
+          'inputs': batch['inputs'].to(DEVICE, non_blocking=True),
+          'targets': batch['targets'].to(DEVICE, non_blocking=True),
+      }
+
+  def _build_dataset(self,
+                     data_rng: spec.RandomState,
+                     split: str,
+                     data_dir: str,
+                     batch_size: int):
+    is_train = split == 'train'
+
+    normalize = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize(
+            mean=[i / 255 for i in self.train_mean],
+            std=[i / 255 for i in self.train_stddev])
+    ])
+    eval_transform_config = normalize
+    train_transform_config = transforms.Compose([
+        transforms.RandomResizedCrop(
+            self.center_crop_size,
+            scale=self.scale_ratio_range,
+            ratio=self.aspect_ratio_range),
+        transforms.RandomHorizontalFlip(),
+        normalize
+    ])
+
+    dataset = CIFAR10(root=data_dir,
+                train=split in ["train","eval_train","validation"],
```

Comment:
Nit: it would be great if you could consistently use single instead of double quotation marks.

Reply:
(And yeah, I did test-run the submission and it is very similar to the imagenet submission code)

Reply:
Cool, what kind of validation performance did you get and after what time?

Reply:
Both PyTorch and Jax models get ~86% accuracy after 60k steps. Few notes:

- Jax trains on entire train partition (50k) whereas pytorch trains on 40k. 
- Jax is able to get in 60k steps in 600 seconds whereas pytorch takes 2400 seconds (i.e. 4x slower) -- I set the number of workers in pytorch dataloader to 4 to get these results... the committed code has `num_workers=0`

Reply:
4x slower is a lot. Do you have an idea where that slowdown is coming from?